llvm-hs-pure/src/LLVM/IRBuilder/Module.hs

```diff
@@ -167,6 +168,62 @@ extern nm argtys retty = do
   let funty = ptr $ FunctionType retty argtys False
   pure $ ConstantOperand $ C.GlobalReference funty nm
 
+-- | An external variadic argument function definition
+externVarArgs 
+  :: MonadModuleBuilder m
+  => Name   -- ^ Definition name
+  -> [Type] -- ^ Parameter types
+  -> Type   -- ^ Type
+  -> m Operand
+externVarArgs nm argtys retty = do
+  emitDefn $ GlobalDefinition functionDefaults
+    { name        = nm
+    , linkage     = External
+    , parameters  = ([Parameter ty (mkName "") [] | ty <- argtys], True)
+    , returnType  = retty
+    }
+  let funty = ptr $ FunctionType retty argtys True
+  pure $ ConstantOperand $ C.GlobalReference funty nm
+
+-- | A global variable definition
+global
+  :: MonadModuleBuilder m
+  => Name       -- ^ Variable name
+  -> Type       -- ^ Type
+  -> C.Constant -- ^ Initializer
+  -> m Operand
+global nm ty initVal = do
+  emitDefn $ GlobalDefinition globalVariableDefaults
+    { name                  = nm
+    , LLVM.AST.Global.type' = ty
+    , linkage               = Weak
+    , initializer           = Just initVal
+    }
+  pure $ ConstantOperand $ C.GlobalReference (ptr ty) nm
+
+-- | Creates a series of instructions to generate a pointer to a string
+-- constant. Useful for making format strings to pass to `printf`, for example
```

Comment:
I think you need to use `@printf@` here, otherwise it will try to reference a Haskell identifier of that name.